Misc/NEWS.d/next/Library/2024-07-16-20-49-07.gh-issue-121804.gYN-In.rst

```diff
@@ -0,0 +1,3 @@
+Add new kwarg ``source`` for
```

Comment:
We don't want to make this argument public so let's not mention it in the change log . Instead, mention what we are fixing in the repl

Reply:
Maybe I should make it underscore-prefixed?

Reply:
Our experience is that if it appears in the signature, people will use it and then changing it will be more challenging. Also some times the underscore prefixed argument is used to avoid collision with keywords or other reserved identifiers.

Reply:
> Our experience is that if it appears in the signature, people will use it and then changing it will be more challenging.

Yeah, I did a proposal to hide this per default: https://discuss.python.org/t/25543.

> Also some times the underscore prefixed argument is used to avoid collision with keywords

I was thinking that it's pep8 violation.  "it is generally better to append a single trailing underscore rather than use an abbreviation or spelling corruption." (c)